pihole/datadog_checks/pihole/pihole.py

```diff
@@ -104,3 +102,73 @@ def check(self, instance):
                 status_code,
             )
             raise Exception('Unexpected response from server')  # if we dont get a response code of '200'
+
+    # Supports generating a SID and CSRF using a pihole web password
+    def _auth_v6_session(self, host, web_password):
+        authUrl = 'http://' + host + '/api/auth'
+        payload = {"password": web_password}
+        json_payload = json.dumps(payload)
+        response = self.http.post(authUrl, data=json_payload)
+        return response.text, response.status_code
+
+    # flattens the returned summary result from raw json to metric format, and prepends the namespace
+    def _flatten_dict(self, d, parent_key='pihole', sep='.'):
+        items = []
+        for k, v in d.items():
+            new_key = f"{parent_key}{sep}{k}"
+            if isinstance(v, dict):
+                items.extend(self._flatten_dict(v, new_key, sep=sep).items())
+            else:
+                items.append((new_key, v))
+        return dict(items)
+
+    # Collect all metrics retunred from the /stats/summary endpoint and submit
+    def _collect_v6_metrics(self, host, headers, custom_tags):
+        statsUrl = 'http://' + host + '/api/stats/summary'
+        stats_resp = self.http.get(statsUrl, headers=headers)
+        flattened = self._flatten_dict(stats_resp.json())
+        for m, v in flattened.items():
+            self.gauge(m, v, custom_tags)
```

Comment:
could we potentially define a list of expected metrics? That way we don't submit anything that is invalid or unknown?